client/src/backend.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Polkadot Client data backend
+
+use state_machine;
+use error;
+use primitives::block;
+use blockchain::{self, BlockId};
+
+/// Block insertion transction. Keeps hold if the inseted block state and data.
+pub trait Transaction {
+	/// Associated state backend type.
+	type State: state_machine::backend::Backend;
+
+	/// Returns pending state.
+	fn state(&self) -> error::Result<Self::State>;
+	/// Append block data to the transaction.
+	fn import_block(&mut self, header: block::Header, body: Option<block::Body>, is_new_best: bool) -> error::Result<()>;
+}
+
+/// Client backend. Manages the data layer.
+pub trait Backend {
+	/// Associated block insetion transaction type.
+	type Transaction: Transaction;
```

Comment:
@gavofyork seems that "Transaction" here refers to a literal database transaction, not a blockchain transaction.

Reply:
hmm... maybe change the name to something that doesn't shadow a rather important concept. Maybe `Action`?

Reply:
I'll go with `BlockImportOperation`